src/internal/m365/onedrive/restore.go

```diff
@@ -710,7 +711,7 @@ func createRestoreFolders(
 			continue
 		}
 
-		folderItem, err := fr.GetFolderByName(ictx, driveID, parentFolderID, folder)
+		folderItem, err := fr.GetFolderByName(ictx, driveID, parentFolderID, folderName)
```

Comment:
this feels redundant. If we continue with using replace to make folders can you add a TODO on this to remove it?

Reply:
it's redundant because we check the cache if we have the folder already, make this request to graph for the folder, and then send a create request with "replace" to make the folder if it doesn't exist. At best we'll send 1 graph request either way (here and find the folder or just outright make the folder). We can remove this if we assume we'll have the create folder code always use replace semantics

Reply:
oh, hah, yes.  monday brain.  Will remove, thank you!

Reply:
no worries, the logic for all the pieces is rather scattered and difficult to pick up on